iceoryx_posh/source/roudi/memory/memory_provider.cpp

```diff
@@ -117,9 +117,9 @@ cxx::expected<MemoryProviderError> MemoryProvider::destroy() noexcept
 
     if (!destructionResult.has_error())
     {
-        rp::BaseRelativePointer::unregisterPtr(m_segmentId);
+        rp::BaseRelativePointer::unregisterPtr(rp::BaseRelativePointer::id_t{m_segmentId});
         m_memory = nullptr;
-        m_size = 0u;
+        m_size = 0U;
```

Comment:
This is also an opportunity to start to get rid of the U suffix.

Reply:
That will be done on a separate PR. Feel free to go for it :grin: